core/frontend/helpers/date.js

```diff
@@ -35,7 +35,7 @@ module.exports = function (date, options) {
     // Documentation: http://momentjs.com/docs/#/i18n/
     // Locales: https://github.com/moment/moment/tree/develop/locale
     const dateMoment = moment(date);
-    dateMoment.locale(themeI18n.locale());
+    dateMoment.locale(options.data.site.locale || themeI18n.locale());
```

Comment:
@ErisDS I'm not sure about this line, but it's the only way I could figure out how to get the tests to work with different locales